### PR TITLE
budget-management

### PR DESCRIPTION
**Pull request by taha-soud**
Changes Made
User Info: We now use a User entity to store user details in the database.

BudgetApp:
Added ways to create users, add money, send money, and find the richest users. All these changes now save in the database.
Testing:

Updated the tests to work with the new database setup.
Folders: 
Added a folder for database settings.

### CODE REVIEW

**Comment by mohammedabualsoud** [comment]
@taha-soud Please change the description to describe  only the changes done within the pull request in very simple words.

File: projects/budget-management/taha-soud-v2/tsconfig.json [hunk 1]
```diff
@@ -25,4 +25,5 @@
   "include": ["src/**/*"],
   /* Exclude node_modules and build directories from compilation */
   "exclude": ["node_modules", "build"]
-}
\ No newline at end of file
+}
```
Comment (mohammedabualsoud) [inline projects/budget-management/taha-soud-v2/tsconfig.json:28]: revert the changes nothing done here

File: projects/budget-management/taha-soud-v2/src/BudgetApp.ts [hunk 2]
```diff
@@ -1,78 +1,109 @@
...
-  public getUsers(): User[] {
-    return this._users;
+  public async getUsers(): Promise<User[]> {
+    const users = await this._userRepository.find();
+    return users;
   }
 
-  public resetUsers(): void {
-    this._users = [];
+  public async resetUsers(): Promise<void> {
```
Comment (mohammedabualsoud) [inline projects/budget-management/taha-soud-v2/src/BudgetApp.ts:105]: why you need this API ? you shouldn't expose API for testing purposed

**Review by mohammedabualsoud — CHANGES_REQUESTED** [review]
@taha-soud Don't use orm in this task please. just use raw queries.

File: projects/budget-management/taha-soud-v2/package.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "taha-v2",
+  "version": "1.0.0",
+  "type": "module",
+  "scripts": {
+    "start": "node build/TestBudgetApp.js"
```
Comment (mohammedabualsoud) [inline projects/budget-management/taha-soud-v2/package.json:6]: this supposed someone already build the app for you. 
Please make it automated like add an entry for building the app you can call it "build" maybe and for start command you should run the build before.

File: projects/budget-management/taha-soud-v2/src/BudgetApp.ts (hunk 2 above)
Comment (taha-soud) [reply projects/budget-management/taha-soud-v2/src/BudgetApp.ts:105]: Because i want the process to be continued, in case the reseting method takes time plus the promise in case a failure happens i want to know

**Comment by taha-soud** [comment]
> @taha-soud Don't use orm in this task please. just use raw queries.



I used it to interact with a relational database directly with js code
I will use raw queries instead
